### PR TITLE
Added virtual property for configuration directory

### PR DESCRIPTION
**Pull request by ghost**
Fix derivative for end user to change the directory of the generated ci configuration file. 
#463 

I confirm that the pull-request:

- [x] Follows the contribution guidelines
- [x] Is based on my own work
- [x] Is in compliance with my employer

### CODE REVIEW

**Comment by matkoch** [comment]
This change really only makes sense with Azure Pipelines. All the other configurations must sit in a specific place. Also, `ConfigurationDirectory` is not used in the base class, which indicates it’s in the wrong place.

**Comment by ghost** [comment]
Thanks for pointing it out, I didn't know some CI have restrictions. I will push a change.

File: source/Nuke.Common/CI/AzurePipelines/AzurePipelinesAttribute.cs [hunk 1]
```diff
@@ -31,8 +31,9 @@ public class AzurePipelinesAttribute : ChainedConfigurationAttributeBase
             _images = new[] { image }.Concat(images).ToArray();
         }
 
+        protected virtual string ConfigurationDirectory => NukeBuild.RootDirectory;
```
Comment (matkoch) [inline source/Nuke.Common/CI/AzurePipelines/AzurePipelinesAttribute.cs]:
```suggestion
        protected virtual AbsolutePath ConfigurationDirectory => NukeBuild.RootDirectory;
```

File: source/Nuke.Common/CI/AzurePipelines/AzurePipelinesAttribute.cs [hunk 2]
```diff
@@ -31,8 +31,9 @@ public class AzurePipelinesAttribute : ChainedConfigurationAttributeBase
...
         private string ConfigurationFileName => _suffix != null ? $"azure-pipelines.{_suffix}.yml" : "azure-pipelines.yml";
-        private string ConfigurationFile => NukeBuild.RootDirectory / ConfigurationFileName;
+        private string ConfigurationFile => (AbsolutePath) ConfigurationDirectory / ConfigurationFileName;
```
Comment (matkoch) [inline source/Nuke.Common/CI/AzurePipelines/AzurePipelinesAttribute.cs]:
```suggestion
        private string ConfigurationFile => ConfigurationDirectory / ConfigurationFileName;
```

Comment (ghost) [reply source/Nuke.Common/CI/AzurePipelines/AzurePipelinesAttribute.cs]: true.

File: source/Nuke.Common/CI/AzurePipelines/AzurePipelinesAttribute.cs (hunk 1 above)
Comment (ghost) [reply source/Nuke.Common/CI/AzurePipelines/AzurePipelinesAttribute.cs]: true.

**Pull request merged**